pants

```diff
@@ -67,6 +101,11 @@ export PY="${PY:-python3}"
 py_major_minor_patch=$(${PY} -c 'import sys; print(".".join(map(str, sys.version_info[0:3])))')
 export PANTS_PYTHON_SETUP_INTERPRETER_CONSTRAINTS="${PANTS_PYTHON_SETUP_INTERPRETER_CONSTRAINTS:-['CPython==${py_major_minor_patch}']}"
 
+# We can currently assume the CI environment does not switch python versions within a shard.
+if [[ "${ONLY_USING_SINGLE_PYTHON_VERSION:-false}" != 'true' ]]; then
+  clean_if_interpreter_constraints_changed
```

Comment:
I liked how you originally passed $PY as an argument to this function, and it assumed nothing about the global env var being defined.

Could we go back to this approach please?
```bash
clean_if_interpreter_constraints_changed "${PANTS_PYTHON_SETUP_INTERPRETER_CONSTRAINTS}"
```

And then change the function's first line to this:
```
current_interpreter_constraints="${1}"
```

Reply:
Yes!